### PR TITLE
Support for terabytes in -X and -XX options

### PR DESCRIPTION
**Pull request by avermeer**
This fix adds support for 'T' and 't' suffix (indicating terabytes)
to -X and -XX options that take  a <size> parameter.

Also mentions that this added support only applies to 64-bit versions
of OpenJ9.

Fixed the sample in 'Using -XX command-line options' which was a -X
based sample and replaced it with an -XX example.

Fixes: #429

### CODE REVIEW

**Comment by avermeer** [comment]
Since last PR: added update of docs/version0.18.md

File: docs/version0.18.md [hunk 1]
```diff
@@ -66,6 +66,9 @@ The Xdump "exit" agent is used in the implementation of `-XX:[+-]ExitOnOutOfMemo
 By caching the Latest User Defined Class Loader (LUDCL), Java applications that use deserialization extensively can see a performance improvement. This
 capability is controlled by the [-Dcom.ibm.enableClassCaching](dcomibmenableclasscaching.md) system property and is now enabled by default. This feature was disabled for the 0.17.0 release due to [issue #7332](https://github.com/eclipse/openj9/issues/7332) which has now been resolved.
 
+### Terabytes suffix support for -X and -XX options that take a size
```
Comment (pshipton) [inline docs/version0.18.md]: Don't forget to add this new topic to the table of contents above, using #terabytes-suffix-support-for-x-and-xx-options-that-take-a-size

**Comment by pshipton** [comment]
Note that here too you should squash the multiple commits and force push the result.

File: docs/version0.18.md (hunk 1 above)
Comment (avermeer) [reply docs/version0.18.md]: @pshipton: done change of table of contents, made commit and squashing !

Comment (pshipton) [reply docs/version0.18.md]: The commits aren't actually squashed.

**Comment by avermeer** [comment]
Completely getting nuts with this one, since the PR leads to "All checks have fails" telling that I failed to include this line in my commits, no matter how many time I put it:
`Signed-off-by: Alexandre Vermeerbergen <avermeerbergen@gmail.com>`
Why newbie committers' life has to be so hard?

**Comment by pshipton** [comment]
The Signed-off-by is missing from the first commit, all the commits need to contain it. The only way to fix it is squashing, where you can ensure the single commit comment contains it.

**Comment by avermeer** [comment]
Okay I have found a git expert in my neighborhood, who finally got the point, which was that when I used  g`it rebase -i <hash>`, I provided the `<hash>` of my latest commit, but my commits were not on the 'main' branch of the branch displayed by `git log --graph --decorate --pretty=oneline --abbrev-commit`, so he said that was my my `git commit --force` lead to nowhere....

He hinted me to provide the <hash> of the first commit following mine on the "main" branch, and YES, it worked !

What is the next step? how can I track when my PR is integrated ?

**Comment by pshipton** [comment]
> What is the next step? how can I track when my PR is integrated ?

The code change needs to be ready to merge (reviewed, approved, PR testing passed), and this needs to be reviewed and approved by the docs team, with all review comments addressed. Once this is merged, it will become part of the docs for the next release, view-able here https://eclipse.github.io/openj9-docs/

**Comment by pshipton** [comment]
@SueChaplain @doveye @pmhayward can someone pls review.

File: docs/xx_jvm_commands.md [hunk 2]
```diff
@@ -27,9 +27,9 @@
 Java&trade; VM command-line options that are specified with `-XX:` are not checked for validity. If the VM does not recognize the option, the option is ignored. These options can therefore be used across different VM versions without ensuring a particular level of the VM.  If you want
 to turn off this behavior to test whether your -XX options are valid, use the [-XX:-IgnoreUnrecognizedXXColonOptions](xxignoreunrecognizedxxcolonoptions.md) option.
 
-For options that take a `<size>` parameter, add a suffix to the size value: "k" or "K" to indicate kilobytes, "m" or "M" to indicate megabytes, or "g" or "G" to indicate gigabytes.
+For options that take a `<size>` parameter, suffix the number with "k" or "K" to indicate kilobytes, "m" or "M" to indicate megabytes, "g" or "G" to indicate gigabytes, or "t" or "T" to indicate terabytes.
```
[the remark this answers is not in the document]
Comment (avermeer) [reply docs/xx_jvm_commands.md]: @pmhayward thank you very much for this review, I'll proceed with the changes.

**Comment by pshipton** [comment]
> I assume that I'm reading various comments correctly and that there are no 32-bit restrictions?

You can't use terabytes on 32-bit, since a number this big isn't supported on 32-bit. The only valid value specified in terabytes is zero. On 32-bit the VM will recognized that a value in terabytes was requested and give an error related to the value being too large (except for zero).

File: docs/xx_jvm_commands.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (avermeer) [reply docs/xx_jvm_commands.md]: @pmhayward again thank you very much for pointing out this oddity in English (hasn't Geroges Orwell predicted that languages will shrink, maybe losing some verbs is the start of it?).

I have committed (if this is still acceptable to consider 'commit' as a verb) the requested change.

File: docs/x_jvm_commands.md [hunk 3]
```diff
@@ -28,7 +28,7 @@
 Use these options to configure the OpenJ9 virtual machine (VM). Unlike standard options, options prefixed with `-X` are nonstandard and are typically unique to a Java&trade; virtual
 machine implementation. However, in some cases, `-X` option names are common to different VM implementations and might have the same function.
 
-For options that take a `<size>` parameter, suffix the number with "k" or "K" to indicate kilobytes, "m" or "M" to indicate megabytes, or "g" or "G" to indicate gigabytes.
```
[the remark this answers is not in the document]
Comment (avermeer) [reply docs/x_jvm_commands.md]: Fixed in my latest commit!

**Comment by pshipton** [comment]
@avermeer there is a conflict that needs to be resolved before this can be merged.

**Comment by avermeer** [comment]
@pshipton thanks for the reminded, I have solved the (trivial) conflict in the TOC file.

**Pull request merged**